python/ray/_private/runtime_env/packaging.py

```diff
@@ -799,18 +833,13 @@ def unzip_package(
         zip_ref.extractall(target_dir)
     if remove_top_level_directory:
         top_level_directory = get_top_level_dir_from_compressed_package(package_path)
-        if top_level_directory is None:
-            raise ValueError(
-                f"The zip package at {package_path} must contain "
-                "a single top level directory. Make sure there "
-                "are no hidden files at the same level as the "
-                "top level directory. You can ensure this by running "
-                "`zip -r example.zip example_dir` from the parent "
-                "directory of example_dir when creating the zip file. "
-                "You can check the contents with `zipinfo -1 example.zip`."
-            )
+        if top_level_directory is not None:
+            # Remove __MACOSX directory if it exists
+            macos_dir = os.path.join(target_dir, MAC_OS_ZIP_HIDDEN_DIR_NAME)
```

Comment:
Should we also remove the `__MACOSX` directory even if there's no top-level directory?

Reply:
I don't have a strong preference, but I would prefer not to, just to change as little possible.